include/zephyr/arch/riscv/arch.h

```diff
@@ -220,6 +220,9 @@ extern void z_irq_spurious(const void *unused);
  */
 static ALWAYS_INLINE unsigned int arch_irq_lock(void)
 {
+#ifdef CONFIG_RISCV_SOC_HAS_CUSTOM_IRQ_LOCK_OPS
+	return z_soc_irq_lock();
```

Comment:
i do not see z_soc_irq_lock defined anywhere

Reply:
this is kinda on purpose. You want to keep the inlining so it is implied to have these functions declared as `static inline` directly in some header file (in my tests I have those in `soc.h` for example).